src/storage-backend.js

```diff
@@ -0,0 +1,848 @@
+let assert = require('assert');
+let crypto = require('crypto');
+let url = require('url');
+let debug = require('debug')(require('path').relative(process.cwd(), __filename));
+let http = require('http');
+
+// We don't follow redirects inside the request library because it does not
+// have a guaruntee that all redirects are secure.  This is a critical part of
+// our system, so we've implemented our own secure redirector
+let request = require('request').defaults({
+  followRedirect: false, 
+});
+let fs = require('fs');
+let stream = require('stream');
+let meter = require('stream-meter');
+let contentDisposition = require('content-disposition');
+let SQSConsumer = require('sqs-consumer');
+
+/**
+ * Return a promise that just waits a certain amoutn of time in ms
+ */
+function delayer (time) {
+  return new Promise(resolve => {
+    setTimeout(resolve, time);
+  });
+}
+
+/**
+ * Obtain a header from a URL.  Do not follow redirects.  Returns the headers,
+ * a caseless to look up header names without worrying about case, the HTTP
+ * status code and the HTTP status message
+ */
+let requestHead = async (u) => {
+  return new Promise((resolve, reject) => {
+    request.head(u, {followRedirect: false}, (err, response) => {
```

Comment:
Just use `http/https` directly... They don't support redirects, oh, and I'm pretty sure headers are always lower-case in node. `http/https` certainly uses lower-case.

Generally, I always try to find a fairly small wrapper around `http/https`, superagent used to be my goto, but now I'm increasingly looking at `got` which seems like a potentially nice module.. It's composed of a lot of smaller modules that are all very specific... Have a look, it could be our superagent replacement.